src/com/goide/runconfig/testing/coverage/GoCoverageProgramRunner.java

```diff
@@ -68,13 +124,46 @@ protected RunContentDescriptor doExecute(@NotNull RunProfileState state, @NotNul
     }
     FileDocumentManager.getInstance().saveAllDocuments();
     CoverageEnabledConfiguration coverageEnabledConfiguration = CoverageEnabledConfiguration.getOrCreate(runConfiguration);
-    runningState.setCoverageFilePath(coverageEnabledConfiguration.getCoverageFilePath());
+    CoveragePatcher runningStatePatcher = new CoveragePatcher(coverageEnabledConfiguration, runConfiguration);
+    runningState.setPatcher(runningStatePatcher);
+
+    if (runConfiguration.getKind() == GoTestRunConfiguration.Kind.DIRECTORY &&
+        !runningStatePatcher.hasPackageCoverageExecutable()) {
+      promptForPackageCoverageTool(runConfiguration);
+    }
 
     ExecutionResult executionResult = state.execute(environment.getExecutor(), this);
     if (executionResult == null) {
       return null;
     }
+    executionResult.getProcessHandler().addProcessListener(new GoCoverageProcessListener(
+        runConfiguration.getDirectoryPath(),
```

Comment:
> I'll see if https://github.com/haya14busa/goverage can be adapted to it. 

Thanks.

> I don't own package-coverage, so it's not something I can change easily either.

Well, you can change Go Plugin for IntelliJ for sure, why do not change package-coverage? ;) 
I think changing `package-coverage` is easier than reimplementing IDE to run a process for each package in IDE.

Anyway, you did a good work and I assure you it won't be thrown away. Even if you close the PR, I'll grab these changes eventually.